app/src/server/tasks/evaluateTestSetEntries.task.ts

```
@@ -0,0 +1,428 @@
+          },
+          {
+            role: "user",
+            content: `Which response is better in the context of the task? Remember to pay attention to the following: ${
+              instructions ?? ""
+            }`,
+          },
+        ],
+        tools: [
+          {
+            type: "function",
+            function: {
+              name: "record_score",
+              parameters: {
+                type: "object",
+                properties: {
+                  explanation: {
+                    type: "string",
+                    description: "An explanation of why you chose the response you did",
+                  },
+                  judgement: {
+                    type: "string",
+                    enum: JUDGEMENT_OPTIONS,
+                    description: "Whose response is better in the context of the task?",
+                  },
+                },
+                required: ["explanation", "judgement"],
+              },
+            },
+          },
+        ],
+      };
+
+      const response = await getOpenaiCompletion(
+        firstResult.datasetEvalDatasetEntry.datasetEntry.dataset.projectId,
+        input,
+      );
+
+      const args = response.choices[0]?.message?.tool_calls?.[0]?.function?.arguments;
+
+      if (!args) throw new Error("No arguments returned" + JSON.stringify(response));
+
+      const parsedArgs = JSON.parse(args);
+
+      if (!parsedArgs["explanation"] || !parsedArgs["judgement"]) {
+        throw new Error("No explanation or judgement returned" + JSON.stringify(response));
+      }
+
+      explanation = parsedArgs["explanation"] as string;
+      judgement = parsedArgs["judgement"] as (typeof JUDGEMENT_OPTIONS)[number];
+
+      if (!JUDGEMENT_OPTIONS.includes(judgement)) {
+        throw new Error("Invalid judgement returned" + JSON.stringify(response));
+      }
+    } catch (e) {
+      console.error("error getting judgement", e);
```

Let's report this error to sentry.